src/playground/sample_data_tooling/sample_data_ingestor/sample_data_ingestor.py

```diff
@@ -0,0 +1,345 @@
+    if number and type(number) is not int or (type(number) is int and number < 0):
+        raise ValueError("number should be a positive integer")
+    if chunk and type(chunk) is not int or (type(chunk) is int and chunk < 0):
+        raise ValueError("chunk should be a positive integer")
+    if timestamp and type(timestamp) is not str:
+        raise TypeError("timestamp should be the field name that contains the timestamp, not the actual unix timestamp")
+    if minutes and type(minutes) is not int or (type(minutes) is int and minutes < 0):
+        raise ValueError("minutes should be a positive integer")
+    if current_date and type(current_date) is not datetime:
+        raise ValueError("current_date should be a datetime object")
+    if max_bulk_size and type(max_bulk_size) is not int or (type(max_bulk_size) is int and max_bulk_size < 0):
+        raise ValueError("max_bulk_size should be a positive integer")
+    if anomaly_detection_trend and type(anomaly_detection_trend) is not list:
+        raise TypeError("anomaly_detection_trend should be a list of config dicts")
+
+
+def build_request_body(index_name:str,
+    file_provided:bool,
+    timestamp:str,
+    minutes:int,
+    chunk:int,
+    current_index:int,
+    dataset:list,
+    max_bulk_size:int
+) -> tuple:
+    """
+    Function that constructs the request body for BULK API call
+
+    Arguments:
+        - index_name: The name of the index to ingest data
+        - file_provided: Boolean flag as to whether or not a file was provided
+        - chunk: How many documents can be ingested per BULK call
+        - timestamp: The field name that contains timestamps
+        - minutes: The time interval for each data point (e.g. if minutes = 2, this tool will generate entries with timestamps that are 2 minutes apart from one another)
+        - current_date: The date at which entries are generated
+        - dataset: The list containing the data to ingest
+        - max_bulk_size: The max amount in bytes of a bulk call
+        - current_index: The index representing the current document in dataset to look at
+
+    Returns:
+        - A tuple containing the request body and the current index to look at
+
+    Raises:
+        - ValueError: current_index should be a positive index position
+        - TypeError: dataset is a list of documents to ingest
+    """
+
+    # First validates input
+    ingest_validation(index_name = index_name,
+        file_provided = file_provided,
+        timestamp = timestamp,
+        minutes = minutes,
+        chunk = chunk,
+        max_bulk_size = max_bulk_size
+    )
+    # Validates function specific input
+    if type(current_index) is not int or (type(current_index) is int and current_index < 0):
+        raise ValueError("current_index should be a positive index position")
+    if type(dataset) is not list:
+        raise TypeError("dataset is a list of documents to ingest")
```

Comment:
is this error statement? is this help "The dataset requires a list of documents to ingest"

Reply:
Fixed the error statement to `The dataset *should be* a list of documents to ingest, not a dict`